deployment/ccip/changeset/sui/operation/ccip_offramp.go

```diff
@@ -0,0 +1,163 @@
+	DestTransferCapId                     string
+	ChainSelector                         uint64
+	PremissionExecThresholdSeconds        uint32
+	SourceChainSelectors                  []uint64
+	SourceChainsIsEnabled                 []bool
+	SourceChainsIsRMNVerificationDisabled []bool
+	SourceChainsOnRamp                    [][]byte
+}
+
+var initializeHandler = func(b cld_ops.Bundle, deps sui_ops.OpTxDeps, input InitializeOffRampInput) (output sui_ops.OpTxResult[DeployCCIPOffRampObjects], err error) {
+	offRampPackage, err := module_offramp.NewOfframp(input.OffRampPackageId, deps.Client)
+	if err != nil {
+		return sui_ops.OpTxResult[DeployCCIPOffRampObjects]{}, err
+	}
+
+	call := offRampPackage.Initialize(
+		bind.Object{Id: input.OffRampStateId},
+		bind.Object{Id: input.OwnerCapObjectId},
+		bind.Object{Id: input.FeeQuoterCapId},
+		bind.Object{Id: input.DestTransferCapId},
+		input.ChainSelector,
+		input.PremissionExecThresholdSeconds,
+		input.SourceChainSelectors,
+		input.SourceChainsIsEnabled,
+		input.SourceChainsIsRMNVerificationDisabled,
+		input.SourceChainsOnRamp,
+	)
+
+	tx, err := call.Execute(b.GetContext(), deps.GetTxOpts(), deps.Signer, deps.Client)
+	if err != nil {
+		return sui_ops.OpTxResult[DeployCCIPOffRampObjects]{}, fmt.Errorf("failed to execute Offramp initialization: %w", err)
+	}
+
+	return sui_ops.OpTxResult[DeployCCIPOffRampObjects]{
+		Digest:    tx.Digest.String(),
+		PackageId: input.OffRampPackageId,
+		Objects:   DeployCCIPOffRampObjects{},
+	}, err
+}
+
+type SetOCR3ConfigInput struct {
+	OffRampPackageId               string
+	OffRampStateId                 string
+	OwnerCapObjectId               string
+	ConfigDigest                   []byte
+	OCRPluginType                  byte
+	BigF                           byte
+	IsSignatureVerificationEnabled bool
+	Signers                        [][]byte
+	Transmitters                   []string
+}
+
+var setOCR3ConfigHandler = func(b cld_ops.Bundle, deps sui_ops.OpTxDeps, input SetOCR3ConfigInput) (output sui_ops.OpTxResult[DeployCCIPOffRampObjects], err error) {
+	offRampPackage, err := module_offramp.NewOfframp(input.OffRampPackageId, deps.Client)
+	if err != nil {
+		return sui_ops.OpTxResult[DeployCCIPOffRampObjects]{}, err
+	}
+
+	call := offRampPackage.SetOcr3Config(
+		bind.Object{Id: input.OffRampStateId},
+		bind.Object{Id: input.OwnerCapObjectId},
+		input.ConfigDigest,
+		input.OCRPluginType,
+		input.BigF,
+		input.IsSignatureVerificationEnabled,
+		input.Signers,
+		input.Transmitters,
+	)
+
+	tx, err := call.Execute(b.GetContext(), deps.GetTxOpts(), deps.Signer, deps.Client)
+	if err != nil {
+		return sui_ops.OpTxResult[DeployCCIPOffRampObjects]{}, fmt.Errorf("failed to execute set ocr3 config in offramp: %w", err)
+	}
+
+	return sui_ops.OpTxResult[DeployCCIPOffRampObjects]{
+		Digest:    tx.Digest.String(),
+		PackageId: input.OffRampPackageId,
+		Objects:   DeployCCIPOffRampObjects{},
+	}, err
+}
+
+var DeployCCIPOffRampOp = cld_ops.NewOperation(
```

Comment:
@stackman27 Is there any reason we need to have Ops in CL Core? If not, I would encourage to have them in `chainlink-sui`, and import them here. It will give us much more flexibility on development

Reply:
that's fair. Removed seq & operations from core repo and move to product specific repo